### PR TITLE
Yaml tests

### LINKED ISSUE #1: Handle malformed YAML more sanely
Instead of throwing 500 on malformed YAML, the site should display the students in the class with correct yaml files.

For malformed yaml it should either:

- Display an alert
- Replace tiles for students with bad YAML files with the message from the YAML parser.

_Migrated from decause/hflossk#569_

### PR DESCRIPTION
**Pull request by liam-middlebrook**
This should fix #1 and #16 

** Please Note: You'll need to edit the auto-generated tox.ini to point to this version of ofCourse and not my branch hosted here on GitHub. I've left it for now so you can confirm that this works. **

### CODE REVIEW

File: ofcourse/test_base/README.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ofCourse Content Repo
+=====================
```
Comment (liam-middlebrook) [inline ofcourse/test_base/README.md]: I'm guessing that @decause or @ryansb will want some more content to go into this file?

**Comment by ryansb** [comment]
I don't like this solution to getting travis for content repos. You don't need a setup.py and tox.ini to run the tests, it's doable with just a `travis.yml` and adding an `ofcourse validate` (or similar) subcommand and seems like a much better approach. I'd rather keep non-actual-content files to a minimum in content repositories. 

Also, this will probably interfere with the way that openshift deployment works.

**Comment by liam-middlebrook** [comment]
Sounds good! I'll move that into `ofcourse validate`

**Comment by liam-middlebrook** [comment]
Okay this should be all good to go now. Once you merge this I think it would be a good time to cut a release. Once you do that we should be able to have the generated TravisCI tests work for any ofcourse instances.

**Comment by ryansb** [comment]
You know you can run those locally with `tox -epep8` right? ;)

**Comment by liam-middlebrook** [comment]
I realized that as soon as I saw the build start. :+1:

**Comment by ryansb** [comment]
Awesome, thanks!

**Pull request merged**